docker/swift-t/Dockerfile

```diff
@@ -10,31 +10,45 @@ FROM 136a88385611
 ENV SWIFT_ROOT=/opt/swift-t
 
 # Install dependencies
-RUN yum -y install mpich-devel # OpenMPI is fine too
+# RUN yum -y install mpich-devel
+RUN yum -y install openmpi-devel
 RUN yum -y install swig
 RUN yum -y install zsh
-RUN wget https://prdownloads.sourceforge.net/tcl/tcl8.6.11-src.tar.gz
+RUN mkdir -pv /tmp/build-tcl
+WORKDIR /tmp/build-tcl
+RUN wget --no-verbose https://prdownloads.sourceforge.net/tcl/tcl8.6.11-src.tar.gz
 RUN tar xfz tcl8.6.11-src.tar.gz
+# Fix Tcl RTLD setting:
+RUN cd tcl8.6.11/unix \
+  && sed -i 90s@^@//@    tclLoadDl.c \
+  && sed -i 92,+2s@^@//@ tclLoadDl.c
 RUN cd tcl8.6.11/unix \
   && ./configure --prefix=/opt/tcl-8.6.11 \
-  && make -j binaries libraries \
+  && make binaries libraries \
   && make install-binaries install-libraries install-headers
```

Comment:
Same thing here.   Sadly the tar should be downloaded, extract, the software configured and installed, and then the tar deleted all in one `RUN` to save layers and avoid image bloat (much like a git repo, docker saves all intermediate state/layers, so leaving the tarball in an intermediate will increase the image size even if its inaccessible in the final image).